src/ServiceControl.Monitoring/Timings/IntervalsStore.cs

```diff
@@ -176,5 +178,28 @@ public class TimeInterval
         internal const int NumberOfHistoricalIntervals = 4 * 5;
 
         static TimeSpan IntervalSize = TimeSpan.FromSeconds(15);
+
+        IDictionary<string, IEnumerable<EndpointInstanceId>> IKnowAboutEndpoints.AllEndpointData()
+        {
+            return intervals.Keys
+                .GroupBy(i => i.EndpointName)
+                .ToDictionary(g => g.Key, g => g.AsEnumerable());
+        }
+
+        protected static MonitoredEndpointValues AggregateTimings(List<EndpointInstanceIntervals> timings)
```

Comment:
Should we move this and `AggregateRetries` to separate static classes calling them `AggregateToValueAverage` and `AggregateToCount`. In the near future we will have message rate which will use the same aggregation as `Retries`